app/metadata.js

```diff
@@ -42,21 +42,27 @@ function getDocumentMetadata(url, window) {
 
 
 function getUrlMetadata(url) {
-  return fetch(url)
-    .then((res) => {
-      if (res.status >= 200 && res.status < 300) {
-        return res;
-      } else {
-        const error = new Error(res.statusText);
-        error.res = res;
-        throw error;
-      }
-    })
-    .then((res) => res.text())
-    .then((body) => getDocumentMetadata(url, domino.createWindow(body)))
-    .catch((err) => {
-      return {};
-    });
+  return new Promise((resolve) => {
+    const result = {url};
+    fetch(url)
+      .then((res) => {
+        if (res.status >= 200 && res.status < 300) {
```

Comment:
I know this is outside the scope of this PR, but what does this validate/do?
Couldn't we get some valid 302s or 304s which would fail this check?